kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionTimeSlotServiceImpl.java

```diff
@@ -27,28 +27,12 @@ public SubscriptionTimeSlotServiceImpl(SubscriptionTimeSlotDataService subscript
     }
 
     @Override
-    public List<SubscriptionTimeSlot> findTimeSlotsForSubscriptionsById(final List<String> subscriptionIds) {
-
-        if (subscriptionIds == null || subscriptionIds.isEmpty()) {
-            LOGGER.warn("No subscription IDs provided, returning empty list.");
-            return new ArrayList<>();
-        }
-
-        StringBuilder idsString = new StringBuilder();
-        int i = 0;
-        while (i < subscriptionIds.size()) {
-            idsString.append("'").append(subscriptionIds.get(i)).append("'");
-            if (i < subscriptionIds.size() - 1) {
-                idsString.append(",");
-            }
-            i++;
-        }
-
+    public List<SubscriptionTimeSlot> findAllTimeSlots() {
         String query = "SELECT ts.subscription_id, ts.subscriptionId, ts.timeStamp1, ts.timeStamp2, ts.timeStamp3 " +
-                "FROM nms_subscriptions_time_slot AS ts " +
-                "WHERE ts.subscriptionId IN (" + idsString + ")";
+                "FROM nms_subscriptions_time_slot ts " +
+                "LEFT JOIN nms_subscriptions s ON ts.subscription_id = s.id;";
 
```

Comment:
This join isn't efficient. Right now, time_slot has little data because it's in the pilot phase, but it will grow to millions soon. We might also get unnecessary old data. Add a WHERE condition to filter only recent data (e.g., last 72 weeks, exclude deactivated or completed ).